tutorials/lesson0-helloworld/hello.template.qnt

```diff
@@ -0,0 +1,319 @@
+    var readByUser: str
+    /*!
+      <step>
+        <file>$file</file>
+        <line>$line</line>
+        <title>Introduce state variables</title>
+        <description>
+The output does not have to be immediately consumed by the user. Hence, we introduce
+another state variable called `readByUser`, which will store the last output read
+by the user. This state variable has the string type too.
+        </description>
+      </step>
+     !*/
+
+    // initialize the state machine that captures the protocol
+    action init = all {
+      consoleOutput' = "",
+      readByUser' = ""
+    }
+    /*!
+      <step>
+        <file>$file</file>
+        <line>$line</line>
+        <title>Introduce an initializer</title>
+        <description>
+Since we are writing distributed protocols in Quint, there is no
+reasonable default for initializing a distributed protocol. This means
+that we have to think about protocol initialization and write an
+initialization action.
+
+This is what we do in the action called `init`. Let's break down
+what is going on here. We have two assignments to state variables:
+
+ 1. `consoleOutput' = ""`
+ 2. `readByUser' = ""`
+
+These two assignments are executed in no particular order. In fact,
+you can imagine that they are executed in any order:
+
+  - 1, then 2
+  - 2, then 1
+  - 1 and 2 at the same time
+
+Once finished, every assignment records the value of its left-hand side
+for the next state, and the assignment unconditionally returns the result `true`.
+Importantly, the assignments do not update the values of `consoleOutput`
+and `readByUser` immediately.
+
+Once both assignments are finished, the operator `all {...}` is finished,
+and it returns `true` (since both assignments returned `true`).
+As a result, the whole action `init` returns the result `true`.
+When `init` is executed as a top-level action, it transitions the state
+machine from whatever state it was in to the state where `consoleOutput = ""`
+and `readByUser = ""`.
+
+If you think that the above behavior of assignments is a bit silly,
+read the next step.
+        </description>
+      </step>
+     !*/
+
+    // write "Hello, world!" in consoleOutput of the state machine,
+    // if the console output is clean
+    action write = all {
+      consoleOutput == "",
+      consoleOutput' = "Hello, world!",
+      readByUser' = readByUser,
+    }
+    /*!
+      <step>
+        <file>$file</file>
+        <line>$line</line>
+        <title>Introduce an action by the computer</title>
+        <description>
+Now it's time to print the "Hello, world!" message. Similar to `init`,
+we introduce the action `write`. In contrast to `init`, this action
+cannot be unconditionally executed in any state of our state machine.
+If you carefully read the description of the action `init` at the previous
+step, you should immediately see that `consoleOutput` changes
+to `"Hello, world!"` in the next state, if `write` returns `true`.
```

Comment:
Is this sentence correct? I can't understand it nor figure out what it was suppose to mean.

Reply:
I will rephrase it